modules/private/internal.go

```diff
@@ -90,5 +82,17 @@ func newInternalRequest(ctx context.Context, url, method string) *httplib.Reques
 			},
 		})
 	}
+
+	if method == "POST" {
+		req.Header("Content-Type", "application/json")
+		if len(postBody) == 1 {
+			jsonBytes, _ := json.Marshal(postBody[0])
+			req.Body(jsonBytes)
+		} else if len(postBody) > 1 {
+			log.Fatal("Too many arguments for newInternalRequest")
+		}
```

Comment:
Hmm…
The current implementation sets a `Content-Type` header for `POST` requests, and then doesn't send anything when `len(postBody) == 0`…

Reply:
What about
```suggestion
	if method == "POST" && len(postBody) == 1 {
		req.Header("Content-Type", "application/json")
		jsonBytes, _ := json.Marshal(postBody[0])
		req.Body(jsonBytes)
	} else if len(postBody) > 1 {
		log.Fatal("Too many arguments for newInternalRequest")
	}
```
instead?

Reply:
I guess the reading new code needs more thoughts.

Sometimes, unfolding the logic/if-block is clearer (to me)

Reply:
I think we can just do fb87fad33ec528dd80a70ffa9c88def09f7a5495 .

That's clear enough. According to HTTP standard, many methods can have "body".

This approach is flexible enough, in the future, we can detect "variable type" to support different "body" types.

```go
	if len(body) == 1 {
		req.Header("Content-Type", "application/json")
		jsonBytes, _ := json.Marshal(body[0])
		req.Body(jsonBytes)
	} else if len(body) > 1 {
		log.Fatal("Too many arguments for newInternalRequest")
	}
```